### PR TITLE
[MLU]support reduce tensors on mlu

### PR DESCRIPTION
**Pull request by kangna-qi**
### PR types
New features

### PR changes
Others

### Describe
support split and concat template class
support strided_memcpy on MLU
support reduce tensors on MLU

### CODE REVIEW

**Comment by kangna-qi** [comment]
测试命令
```
./test_group
```

测试日志

File: paddle/fluid/imperative/tests/test_group.cc [hunk 1]
```diff
@@ -180,5 +182,19 @@ TEST(TestGroup, TestXPUConcatSplit) {
 }
 #endif
 
+#if defined(PADDLE_WITH_MLU)
```
Comment (qili93) [inline paddle/fluid/imperative/tests/test_group.cc]: 这里需要改成 PADDLE_WITH_CNCL 不是 PADDLE_WITH_CNCL

Comment (kangna-qi) [reply paddle/fluid/imperative/tests/test_group.cc]: 已修改

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**